### PR TITLE
feat: add routes for recovering tr funds

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/4386335792).

https://user-images.githubusercontent.com/1618764/224334762-3bfb5c8f-c125-4b95-b1b7-4a760dcca833.mp4

<br />

Tx is slow to show up in mempool at the end, but gets there eventually.

### CODE REVIEW

File: src/app/features/retrieve-taproot-to-native-segwit/retrieve-taproot-to-native-segwit.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+          // eslint-disable-next-line no-console
+          console.log(resp);
+          toast.success('Transaction broadcasted succesfully');
+          await delay(1000);
+          navigate(RouteUrls.Home);
+          void analytics.track('broadcast_retrieve_taproot_to_native_segwit');
+          setIsLoading(false);
+        }}
+      >
+        Retrive bitcoin
```
Comment (fbwoolf) [inline src/app/features/retrieve-taproot-to-native-segwit/retrieve-taproot-to-native-segwit.tsx]:
```suggestion
        Retrieve bitcoin
```

File: src/app/routes/app-routes.tsx [hunk 2]
```diff
@@ -226,6 +230,9 @@ function AppRoutesAfterUserHasConsented() {
         <Route path={RouteUrls.Unlock} element={<Unlock />}>
           {settingsModalRoutes}
         </Route>
+        <Route path={RouteUrls.Unlock} element={<Unlock />}>
```
Comment (fbwoolf) [inline src/app/routes/app-routes.tsx]: Did you mean to duplicate this?

File: src/app/features/retrieve-taproot-to-native-segwit/retrieve-taproot-to-native-segwit.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    if (!feeRate) return createMoney(0, 'BTC');
+    const txSizer = new BtcSizeFeeEstimator();
+    const { txVBytes } = txSizer.calcTxSize({
+      input_count: uninscribedUtxos.length,
+      p2wpkh_output_count: 1,
+    });
+    return createMoney(Math.ceil(txVBytes * feeRate.hourFee), 'BTC');
+  }, [feeRate, uninscribedUtxos.length]);
+
+  async function generateRetrievalTransaction() {
```
Comment (fbwoolf) [inline src/app/features/retrieve-taproot-to-native-segwit/retrieve-taproot-to-native-segwit.tsx]: Maybe pull these out into a separate hook file?

**Review by fbwoolf — APPROVED** [review]

File: src/app/components/info-card/info-card.tsx [hunk 4]
```diff
@@ -27,15 +20,15 @@ export function InfoCard({ children }: InfoCardProps) {
 // InfoCardRow
 interface InfoCardRowProps {
   title: string;
-  value: string;
+  value: React.ReactNode;
```
Comment (alter-eggo) [inline src/app/components/info-card/info-card.tsx:23]: `value: React.ReactNode | string`

File: src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts [hunk 5]
```diff
@@ -115,3 +115,26 @@ export function useSignBitcoinNativeSegwitTx() {
     [keychain]
   );
 }
+
+// export function useCurrentAccountNativeSegwitSigner() {
```
Comment (alter-eggo) [inline src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts]: should it be here commented?

Comment (alter-eggo) [reply src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts]: its commented to avoid `unused exports` rule?

**Review by alter-eggo — APPROVED** [review]

File: src/app/components/info-card/info-card.tsx (hunk 4 above)
Comment (kyranjamie) [reply src/app/components/info-card/info-card.tsx:23]:
```
    //
    // React Nodes
    // http://facebook.github.io/react/docs/glossary.html
    // ----------------------------------------------------------------------

    type ReactText = string | number;
    type ReactChild = ReactElement | ReactText;

    interface ReactNodeArray extends Array<ReactNode> {}
    type ReactFragment = {} | ReactNodeArray;
    type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
```

ReactNode includes `string` as an allowed type so didn't think this was necessary

**Comment by markmhendrickson** [comment]
The amount here seems to capture inbound pending amounts. Not the biggest blocker, but ideally we exclude those / only show settled minus not outbound pending?

Screenshot 2023-03-10 at 16 28 29

File: src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts (hunk 5 above)
Comment (kyranjamie) [reply src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts]: ah yeah, meant to remove

**Comment by fbwoolf** [comment]
🚀 Looks good!

**Pull request merged**